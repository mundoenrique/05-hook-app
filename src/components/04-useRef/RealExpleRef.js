import { useState } from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

export default function RealExpleRef() {
	const [show, setShow] = useState(false);

	return (
		<>
			<h3>RealExpleRef</h3>
			<button className="btn btn-info" onClick={() => setShow(!show)}>
				Mostrar/Ocultar
			</button>
			{show && <MultipleCustomHooks />}
		</>
	);
}
