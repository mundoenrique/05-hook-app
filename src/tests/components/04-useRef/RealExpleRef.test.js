import { shallow } from 'enzyme';
import RealExpleRef from '../../../components/04-useRef/RealExpleRef';

describe('Pruebas sobre el componente <RealExpleRef />', () => {
	const wrapper = shallow(<RealExpleRef />);
	test('Debe renderizar correctamente el componente <RealExpleRef />', () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
	});

	test('Debe renderizar correctamente el componente <MultipleCustomHooks />', () => {
		wrapper.find('.btn-info').simulate('click');

		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
	});
});
