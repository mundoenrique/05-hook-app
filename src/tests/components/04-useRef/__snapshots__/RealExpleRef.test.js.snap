// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <RealExpleRef /> Debe renderizar correctamente el componente <RealExpleRef /> 1`] = `
<Fragment>
  <h3>
    RealExpleRef
  </h3>
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Mostrar/Ocultar
  </button>
</Fragment>
`;
